// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";


package chat;

service ChatService {
  // bidirectional streaming RPC: both client and server can send messages
  rpc Chat(stream ChatRequest) returns (stream ChatResponse) {};
}

message ChatRequest {
  string user_id = 1;
  string text = 2;
}

message ChatResponse {
  string user_id = 1;
  string text = 2;
}